source/particle/world.cc

```diff
@@ -35,6 +35,57 @@ namespace aspect
 {
   namespace Particle
   {
+    namespace
+    {
+      /**
+       * This function is used as comparison argument to std::sort to sort the
+       * vector of tensors @p center_directions by its scalar product with the
+       * @p particle_direction tensor. The sorted indices allow to
+       * loop over @p center_directions with increasing angle between
+       * @p particle_direction and @p center_directions.
+       */
+      template <int dim>
+      bool
+      compare_particle_association(const unsigned int a,
+                                   const unsigned int b,
+                                   const Tensor<1,dim> &particle_direction,
+                                   const std::vector<Tensor<1,dim> > &center_directions)
```

Comment:
I assume the vectors in the last two arguments are normalized to length 1? If so maybe add that this function assumes so.